great_expectations/core/expectation_suite.py

```diff
@@ -218,13 +212,15 @@ def _process_expectation(
     @public_api
     def delete_expectation(self, expectation: Expectation) -> Expectation:
         """Delete an Expectation from the collection.
+        The input Expectation must be in the suite and referenced by index.
```

Comment:
Hmm this is fair - I want to encourage users to delete by using `suite.expectations[idx]` instead of `gxe.ExpectWhateverWhatever`. I think the example here is good enough.

Reply:
Yeah, the example is fine with me. FWIW I'd personally probs find it by like `next(e for i in suite.expectations where isinstance(e, MyExpectation))`